### ISSUE TITLE
BUG: <The method of percentile function may be wrong.>

### ISSUE THREAD

**Issue by Leo-Ji2020** [numpy/numpy#26379]
### Describe the issue:

Hi, developers,
       I found the **numpy.percentile()** function may be wrong. My numpy version is 1.23.5.
       Here is a reference video from Khan Academy: https://www.youtube.com/watch?v=qLYYHWYr8xI
       Let's say we have an array A=[1, 2, 2, 3, 3, 4, 5, 5, 7, 9]. There are 10 numbers in it. According to the Khan Academy, the Q1 should be the median of the left half of the array [1,2,2,3,3], while the Q3 should be the median of the right half of the array [4,5,5,7,9]. Therefore, the results should be Q1=2, Q3=5. Then the IQR should be Q3-Q1=3.
       However, when using numpy.percentile() function, i got the result of Q1=numpy.percentile(A, 25)=2.25, Q3=numpy.percentile(A, 75)=5.0. Then the IQR = 5-2.25=2.75. (The IQR can be calculated using scipy.stats,iqr(A)=2.75). 
       I don't know how the Q1=2.25 when using numpy to do the calculation.

       Furthermore, when the number of array is odd, let's say we have an array B=[41, 42, 52, 55, 60, 61, 61, 65, 68]. According to the Khan Academy, the median value is 60. Then, Q1 should be the left half of the array without the median value [41, 42, 52, 55], while the Q3 should be the median of the right half of the array without the median value [61, 61, 65, 68]. Therefore, the results should be Q1=(42+52)/2=47, Q3=(61+65)/2=63. Then the IQR should be Q3-Q1=63-47=16.
       However, the  numpy.percentile() function gave the results as Q1=numpy.percentile(B, 25)=52, Q3=numpy.percentile(B, 75)=61. Then the IQR =Q3-Q1=9. These results are quite different from the calculation in the traditional way. 
 
       Could you please check the method of numpy.percentile()? Since the Khan Academy is one of the famous math education website of the world, the difference method of the calculation between numpy and Khan Academy may cause difficults for students to learn the math.
       Thank you!

Best wishes,

Dr. Lei Ji

### Reproduce the code example:

```python
import numpy as np

A=[1, 2, 2, 3, 3, 4, 5, 5, 7, 9]
B=[41, 42, 52, 55, 60, 61, 61, 65, 68]

Q1A, Q3A = np.percentile(A, 25), np.percentile(A, 75)
IQRA = Q3A - Q1A

Q1B, Q3B = np.percentile(B, 25), np.percentile(B, 75)
IQRB = Q3B - Q1B
```

### Python and NumPy Versions:

numpy: 1.23.5, 
python: 3.9.13

### Runtime Environment:

windows 10

**Comment by seberg** [comment]
There are many ways to define to quantiles, please see the notes in our docs:
https://numpy.org/devdocs/reference/generated/numpy.percentile.html or e.g. wikipedia.
The differences are contiguous/non-contiguous estimates as well as how to account for estimating it based on a representative sample rather than the full data.

Presumably, one of our methods matches what that video uses (I don't recall if NumPy 1.23 had all of the methods, the old ones were not a good choice comparing to literature).

(There is some ongoing work to make the docs a bit more clear.)

---

Thanks, but I am going to close this, I would think one of the methods matches (on a new one).  If not, please let us know (although it would be good which method is potentially wrong and why).

**Comment by Leo-Ji2020** [comment]
Thank you for your reply. Your suggestion is very valuable.On May 3, 2024, at 1:57 AM, Sebastian Berg ***@***.***> wrote:﻿
Closed #26379 as not planned.
